src/main/java/me/skulduggerry/gameoflife/controller/GenerationController.java

```diff
@@ -26,32 +26,38 @@
 
     @NonNull
     private final ObjectMapper mapper;
+    @NonNull
+    private final GenerationService generationService;
+    @NonNull
+    private final GenerationConverter generationConverter;
+
+    @PostMapping(value = UPLOAD_PATH, consumes = "application/json", produces = "application/json")
+    public String upload(@RequestBody TransferGeneration transferGeneration) {
+        log.debug("Uploaded a valid file");
+        log.debug("Send json:\n" + transferGeneration.toString());
+        generationConverter.toWorkGeneration(transferGeneration);
+
+        return """
+                {
+                    "upload": "success"
+                }
+                """;
+    }
+
+    @GetMapping(NEXT_GENERATION_PATH)
+    public TransferGeneration getNextGeneration() {
+        log.debug("Request for next generation.");
+        WorkGeneration currentGeneration = generationService.getNextGeneration();
+        TransferGeneration transferGeneration = generationConverter.toTransferGeneration(currentGeneration);
+        log.debug("Send json:\n" + transferGeneration.toString());
+        return transferGeneration;
+    }
 
-    @PostMapping(UPLOAD_PATH)
-    public String upload(@RequestPart("file") MultipartFile file, HttpServletResponse response) throws IOException {
-        try {
-            response.sendRedirect("/");
-        } catch (IOException e) {
-            throw new RedirectionFailedException(e);
-        }
-
-        if (file.isEmpty()) {
-            log.info("Uploaded file is empty");
-            return """
-                    {
-                        "upload": "empty"
-                    }
-                    """;
-        } else {
-            log.info("Uploaded a valid file");
-
-            Generation generation = mapper.readValue(file.getInputStream(), Generation.class);
-            log.info(generation.toString());
-            return """
-                    {
-                        "upload": "success"
-                    }
-                    """;
-        }
+    @GetMapping(CURRENT_GENERATION_PATH)
+    public TransferGeneration getCurrentGeneration() {
+        log.debug("Request for current generation.");
+        TransferGeneration transferGeneration = generationConverter.toTransferGeneration(WorkGeneration.getInstance());
```

Comment:
Access to` WorkGeneration` should be encapsuled in a separate place - I suggest `GenerationService`